jhu-idc/idc-isle-dc
# Updates Audio and Video Conditions

Resolves: https://github.com/jhu-idc/iDC-general/issues/427 

This PR adds a Condition that cause service files to only be created for Audio and Video files in the event that they don't already have one. If an ingest marks a file as "Original File" and "Service File", then a service file derivative will not be made for the file. 

To test:
 - check out and run this branch
 - For Video: 
     - Create a node with model set to Video.  Add a mp4 file to it as its media.  Make sure the file is only marked as "Original File".  After saving you should see that a derivative Service File gets created. 
     - Create a node with model set to Video. Add a mp4 file to it as its media. Make sure the file is marked as both "Original File" and "Service File".  After saving you should see that no Service File derivative is created for it. 
     - In both cases, make sure that the video player is present and functional on the node's view page. 
 - For Audio:
     - Create a node with model set to Audio. Add a mp3 file to it as its media. Make sure the file is only marked as "Original File". After saving you should see that a derivative Service File gets created. 
     - Create a node with model set to Audio. Add a mp3 file to it as its media. Mark this file as both Original File and Service File. After saving you should see that no service file is generated for it. 
     - In both cases, ensure that the audio player is available and functional on the page. 

In both scenarios (Audio / Video) look at the logs as you run these and ensure that Homarus doesn't run in the case you don't expect it to, and that it does run in the cases you do. 

Ensure that the Video files both have thumbnails & FITS files still created for them, whether or not they have a service file deriv.  And ensure that the Audio files do have FITS files generated as well (no thumbnails). 

Notes:
- The `make config-export` touched a lot of the context files - this is because I reordered them. 
- This PR relies on a few new assets in the migration assets container. PR for that here:https://github.com/jhu-idc/migration-file-server/pull/3.  I did push another asset image, so this PR should work. 
- I took the opportunity to move an earlier condition I created in the islandora_defaults module and put it into the idc_defaults module.
